core-bundle/src/Command/MigrateCommand.php

```diff
@@ -684,10 +685,18 @@ private function validateDatabaseVersion(bool $asJson): bool
             return true;
         }
 
-        $message = sprintf('Wrong database version configured, please set it to "%s"', $version);
-
         if ($currentVersion = $this->connection->getParams()['serverVersion'] ?? null) {
-            $message .= sprintf(', currently set to "%s"', $currentVersion);
+            $message =
+                <<<EOF
+                    Wrong database version configured!
+                    You have version $version but the database connection is configured to $currentVersion.
+                    EOF;
+        } else {
+            $message =
+                <<<EOF
+                    Wrong database version configured!
+                    You have version $version but the database connection is configured to a different version.
```

Comment:
```suggestion
                    Cannot detect database version!
                    Your database version could not be detected automatically. Please check you database configuration.
```

It does not have to be the exact wording, just something that describes the situation more accurately. 😄

Reply:
huh? that message has a completely different meaning?

Reply:
I think this else branch can never be reached anyways because if `'serverVersion'` is not set, doctrine would run the same auto-detection that we run and would detect the same version anyway.

So it would probably be better to default to `""` or so for this rare edge-case.
Remove the `if (…)` completely and use something like this?

```php
$currentVersion = $this->connection->getParams()['serverVersion'] ?? '""';
```

Reply:
Exactly, because the original message is wrong. If `getParams()['serverVersion']` is `null`, the database has not been configured at all, therefore it is not correct to say _is configured to a different version_.

Reply:
I tried to keep the meaning of the existing message. But if the serverVersion is not configured, shouldn't we handle this as an error? Because Doctrine would retrieve the version on every page load?

Reply:
I don’t think we should force users to set it. This check should only validate that there is not a wrong version configured.

Reply:
Then we should change the whole method in Contao 4.9 and upstream?

Reply:
Yes, after you finished this PR and it was merged you should create a backport for 4.9.